HLTrigger/Configuration/python/HLT_25ns15e33_v4_cff.py

```diff
@@ -71340,9 +71405,11 @@
 fragment.AlCa_EcalEtaEEonly_v8 = cms.Path( fragment.HLTBeginSequence + fragment.hltL1sAlCaEcalPi0Eta + fragment.hltPreAlCaEcalEtaEEonly + fragment.HLTDoFullUnpackingEgammaEcalSequence + fragment.hltSimple3x3Clusters + fragment.hltAlCaEtaRecHitsFilterEEonlyRegional + fragment.hltAlCaEtaEEUncalibrator + fragment.hltAlCaEtaEERechitsToDigis + fragment.HLTEndSequence )
 fragment.HLTriggerFinalPath = cms.Path( fragment.hltGtStage2Digis + fragment.hltScalersRawToDigi + fragment.hltFEDSelector + fragment.hltTriggerSummaryAOD + fragment.hltTriggerSummaryRAW + fragment.hltBoolFalse )
 fragment.HLTAnalyzerEndpath = cms.EndPath( fragment.hltGtStage2Digis + fragment.hltPreHLTAnalyzerEndpath + fragment.hltL1TGlobalSummary + fragment.hltTrigReport )
+fragment.ScoutingCaloOutput = cms.Path( fragment.hltGtStage2Digis + fragment.hltPreScoutingCaloOutput + fragment.hltFEDSelectorL1 + fragment.hltScoutingCaloPacker )
+fragment.ScoutingPFOutput = cms.Path( fragment.hltGtStage2Digis + fragment.hltPreScoutingPFOutput + fragment.hltFEDSelectorL1 + fragment.HLTPFScoutingPackingSequence )
```

Comment:
I have done this PR "as a backport" (meaning, trying not to do new things), but I'm wondering if this change works (it might just be my ignorance of the framework).

Taking `hltScoutingCaloPacker` as an example, it consumes products that are not in the Path `ScoutingCaloOutput`, and `HLTScoutingCaloProducer` will not complain about missing collections (it just produces empty/dummy collections if no inputs are found). The intention would be that `hltScoutingCaloPacker` 'waits' for those collections, but my understanding is that, if more than one thread is used (which is likely the case in production), there is no guarantee on the order in which the Paths run, so in general this would not work (it may work with 1 single thread, because the Paths are executed in order, and the Paths with the Scouting packers are towards the end of the Schedule). It can also be that (1) my understanding is wrong, and/or (2) `8_0_X` behaves differently from recent cycles in this respect.

@makortel, could you please clarify (and correct any of the above) ?

Reply:
Okay, thanks for clarifying (my understanding was wrong).

In case it was unclear, the desired behaviour is that `hltScoutingCaloPacker` runs only after all the products it consumes ran (my doubt came from the fact that those products are created by Paths different from the one including `hltScoutingCaloPacker`).

To see if I understand

> since 9_0_0_pre1, all (End)Paths are processed concurrently (https://github.com/cms-sw/cmssw/pull/15882) in non-deterministic order, such that the execution of modules in the Paths follow the order of data dependencies (i.e. hltScoutingCaloPacker is run only after all its input producers either have been run, or known not to be run because of the filtering)

this (esp. "that the execution of modules in the Paths follow the order of data dependencies") means that this approach leads to the desired behaviour in those release cycles (e.g. #21297 for `10_0_X` produces the desired behaviour).

> Earlier (i.e. in 8_0_X) framework processes the (End)Paths in the order specified by the Schedule. If the Schedule definition below has all the Paths of the input producers before the Path(s?) that contains hltScoutingCaloPacker,

yes, this is the case (there is only 1 Path with `hltScoutingCaloPacker`)

> the job should behave as you'd expect (or how I understood what you'd expect).

this means this PR happens to create the desired behaviour for 8_0_X.

Reply:
> To see if I understand
> 
> > since 9_0_0_pre1, all (End)Paths are processed concurrently (#15882) in non-deterministic order, such that the execution of modules in the Paths follow the order of data dependencies (i.e. hltScoutingCaloPacker is run only after all its input producers either have been run, or known not to be run because of the filtering)
> 
> this (esp. "that the execution of modules in the Paths follow the order of data dependencies") means that this approach leads to the desired behaviour in those release cycles (e.g. #21297 for `10_0_X` produces the desired behaviour).

Correct.